python/ruff/__main__.py

```diff
@@ -3,6 +3,23 @@
 import sysconfig
 from pathlib import Path
 
+RUFF_PATHS = [
+    Path(sysconfig.get_config_var("userbase")) / "bin" / "ruff",
```

Comment:
I expect this is wrong on Windows. The right way to get the `--user` scripts path is `sysconfig.get_path("scripts", scheme=sysconfig.get_preferred_scheme("user"))`—but that requires Python ≥ 3.10, so we’ll need to figure out the right translation for older Python.

Reply:
Can you think of any other projects that've had to solve this problem?